### PR TITLE
`colorMode` key improvements; `computed` dependencies update

### PR DESCRIPTION
**Pull request by thompsongl**
### Summary

Again, more for discussion than to merge this in its current state. Let's see if the changes are closer to your ideal API

### CODE REVIEW

File: src/global_styling/variables/_size.ts
Comment (cchaos) [inline src/global_styling/variables/_size.ts]: I'm still confused by this first parameter. If it's always an empty array, can this not just be the default and sent to the end of the params list?

Comment (thompsongl) [reply src/global_styling/variables/_size.ts]: Sure, we could swap the params

Comment (thompsongl) [reply src/global_styling/variables/_size.ts]: Swapped params; second is optional

**Comment by thompsongl** [comment]
~TS-to-PropTypes parsing has errors. Trying to figure it out.~

^ is happening in the target branch, so this does not need to be solved in this branch. It will likely be difficult to resolve before there is a clearer theme shape, and themes (default, ams) match.

File: src-docs/src/views/emotion/canopy.tsx
Comment (cchaos) [inline src-docs/src/views/emotion/canopy.tsx]: Just double-checking that this doesn't actually need to be `computed` because it seems to just be returning a straight hex value?
```suggestion
        success: '#85E89d',
```

File: src/global_styling/variables/_borders.ts
Comment (cchaos) [inline src/global_styling/variables/_borders.ts:34]: Is this a possible different way to write this, do does the array values have to be a single key?
```suggestion
  thin: computed(([border]) => `${border.widthThin} solid ${border.color}`, ['border']),
```

Comment (cchaos) [reply src/global_styling/variables/_borders.ts:34]: Also could we simplify further to allow the second param to be a single value as well as an array like:

```suggestion
  thin: computed((border) => `${border.widthThin} solid ${border.color}`, 'border'),
```

File: src/global_styling/variables/_colors.ts
Comment (cchaos) [inline src/global_styling/variables/_colors.ts:37]: This might have answered my question above. Just want to know what the preferred guidance is for these computed values. Is one method better than the other?

File: src/global_styling/variables/_size.ts
Comment (cchaos) [inline src/global_styling/variables/_size.ts:34]: This is fantastic! So much simpler! Is this then even something a consumer could do?

File: src/global_styling/variables/_typography.ts
Comment (cchaos) [inline src/global_styling/variables/_typography.ts]: Yeah so we talked about maybe for the color mode keys, they're uppercased like `LIGHT` and `DARK` leaving any lowercase versions allowed as normal keys here. But I'm fine with this for now.

**Review by cchaos — APPROVED** [review]
Thanks!! A lot of the consuming syntax is so simple and intuitive. My following comments are mostly for my clarification. Go ahead and merge this when you're happy with it and/or want to address anything as a follow up.

File: src-docs/src/views/emotion/canopy.tsx
Comment (thompsongl) [reply src-docs/src/views/emotion/canopy.tsx]: Nope. Just left over from some check I was doing.

File: src/global_styling/variables/_borders.ts
Comment (thompsongl) [reply src/global_styling/variables/_borders.ts:34]: > Is this a possible different way to write this, do does the array values have to be a single key?

This is similar to object spreading, and I haven't been able to make it work yet. It's on my radar, though, and I'll keep exploring options.

> Also could we simplify further to allow the second param to be a single value as well as an array like

Sure, we could allow single dependency values in addition to arrays.

File: src/global_styling/variables/_colors.ts
Comment (thompsongl) [reply src/global_styling/variables/_colors.ts:37]: It's entirely up to consumer preference. I'm not sure there's a large enough computational difference to say that one is better than the other.

File: src/global_styling/variables/_size.ts
Comment (thompsongl) [reply src/global_styling/variables/_size.ts:34]: Yep, consumers can use utility functions like this if we export them, or they can write their own curried/partial functions to do similar things.

File: src/global_styling/variables/_typography.ts
Comment (thompsongl) [reply src/global_styling/variables/_typography.ts]: I'll change this back before merging. I think it's certain we won't use `light` and `dark` as color mode keys, so moving to something else right now makes sense.

**Comment by thompsongl** [comment]
Gonna make a couple changes before merging this

**Comment by thompsongl** [comment]
All done.  Apparently only you can merge PRs here

**Pull request merged**
